pkg/tsdb/azuremonitor/types/types.go

```diff
@@ -139,17 +140,27 @@ type AzureMonitorJSONQuery struct {
 // AzureMonitorDimensionFilter is the model for the frontend sent for azureMonitor metric
 // queries like "BlobType", "eq", "*"
 type AzureMonitorDimensionFilter struct {
-	Dimension string `json:"dimension"`
-	Operator  string `json:"operator"`
-	Filter    string `json:"filter"`
+	Dimension string   `json:"dimension"`
+	Operator  string   `json:"operator"`
+	Filters   []string `json:"filters,omitempty"`
+	// Deprecated: To support multiselection, filters are passed in a slice now. Also migrated in frontend.
+	Filter *string `json:"filter,omitempty"`
 }
 
-func (a AzureMonitorDimensionFilter) String() string {
-	filter := "*"
-	if a.Filter != "" {
-		filter = a.Filter
+func (a AzureMonitorDimensionFilter) ConstructFiltersString() string {
+	if len(a.Filters) == 1 {
+		return fmt.Sprintf("%v %v '%v'", a.Dimension, a.Operator, a.Filters[0])
+	} else {
```

Comment:
I don't think you need this `if` (the result would be the same with the `else` logic as far as I can see)

Reply:
Yep will update this 👍